optuna/integration/catalyst.py

```diff
@@ -0,0 +1,61 @@
+import optuna
+
+
+if optuna.type_checking.TYPE_CHECKING:
+    from typing import Any  # NOQA
```

Comment:
[MUST] type_checking is not necessary in this case. The motivation for the use of type_checking.TYPE_CHECKING constraint is avoiding a circular module dependency in this repo.

There are similar codes in Optuna. For example https://github.com/optuna/optuna/blob/master/optuna/integration/lightgbm_tuner/alias.py#L1-L6. This code is also inappropriate. Sorry for confusing you.

```suggestion
from typing import Any  # NOQA
```